### DISCUSSION TITLE
What is ESP3DLib compared to ESP3D ?

### DISCUSSION THREAD

**Discussion by luc-github** [luc-github/ESP3DLib#19] [Q&A]
* ESP3D is the original project it is Firmware and it target ESP32 and ESP8266 boards used as daughter board of printer boards, it need a serial connection between ESP board and printer board to work 

* ESP3DLib is a library to be used by Marlin Firmware only (for the moment), Marlin can now be installed on ESP32 boards as main MCU, so it become a part of printer FW itself. it is not for extern boards

**Comment by hyzx86** [comment]
Sorry, I did not understand this project too much, let me explain my understanding:

Does the ESP3DLib project use ESP32 as the core to replace the Arduino Mega motherboard? ESP32 directly drives the stepper motor? But I did not find any pin definitions in the project.

or

Does ESP3DLib use ESP32 to directly read the Gcode file on its own SD card, and then send commands to the Marlin printer?

**Reply by luc-github** [reply]
first statement: esp3dlib is a library used by Marlin firmware - all pinout are there (Marlin FW)
